thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/model/ThreadModel.kt

```diff
@@ -3,260 +3,283 @@ package io.github.paulgriffith.kindling.thread.model
 import com.formdev.flatlaf.extras.FlatSVGIcon
 import io.github.paulgriffith.kindling.utils.Column
 import io.github.paulgriffith.kindling.utils.ColumnList
-import io.github.paulgriffith.kindling.utils.firstNotNull
 import org.jdesktop.swingx.renderer.DefaultTableRenderer
 import org.jdesktop.swingx.renderer.StringValues
-import java.text.DecimalFormat
+import java.lang.Thread.State.BLOCKED
+import java.lang.Thread.State.NEW
+import java.lang.Thread.State.RUNNABLE
+import java.lang.Thread.State.TIMED_WAITING
+import java.lang.Thread.State.WAITING
+import java.text.NumberFormat
 import javax.swing.table.AbstractTableModel
+import java.lang.Thread.State as ThreadState
 
 // A thread's lifespan across multiple thread dumps
 typealias ThreadLifespan = List<Thread?>
-fun ThreadLifespan.toThread(): Thread = firstNotNull()
+
+sealed class ThreadColumnList : ColumnList<ThreadLifespan>() {
```

Comment:
sealed class + use of Column pseudo-constructor makes it possible to re-use columns in the different column lists below